tests/fixtures/studies/test_params/analysis/study_definition.py

```diff
@@ -0,0 +1,24 @@
+from cohortextractor import StudyDefinition, params, patients
+
+age_suffix = params["age_suffix"]
+sex_suffix = params["sex_suffix"]
```

Comment:
If an invalid parameter is passed to `generate_cohort`, will it get this far and raise a KeyError when the study def is loaded?  Will that get caught in a sensible way?

Reply:
Also a good point. I've made `params` a custom dict subclass which raises a custom error in place of `KeyError` which is then caught by our usual error handling.